Cast geometry from array

trin4ik opened the pull request:
At now we can do `$serialize = Model::find(35)->toArray();` but cant use `$model = new Model($serialize)`. This PR look to values and try to use `Geometry::fromJson()` to serialized array.

MatanYadaev commented on src/GeometryCast.php:
```diff
@@ -62,6 +62,10 @@ public function set($model, string $key, $value, array $attributes): Expression|
     }
 
+    if (is_array($value) && isset($value['type']) && $value['type'] === class_basename($this->className)) {
+        $value = Geometry::fromJson(json_encode($value));
```
Please add a `Geometry@fromArray` method that does the same logic and use it here instead.

MatanYadaev reviewed (changes requested):
Thanks for the PR. Please add tests. This package should be 100% covered with tests.

trin4ik commented:
hi, tests added, method `fromArray` created

MatanYadaev commented on src/Objects/Geometry.php:
```diff
@@ -106,26 +107,41 @@ public static function fromWkt(string $wkt, int $srid = 0): static
+     */
+    public static function fromArray(array $geometry): static
+    {
+        if (! isset($geometry['type'])) {
+            throw new OutOfBoundsException('Your geometry doesn\'t contain "type"');
+        }
```
I believe this validation isn't necessary, we already validate for correctness of GeoJson here: https://github.com/MatanYadaev/laravel-eloquent-spatial/blob/master/src/Factory.php#L34

MatanYadaev commented on src/Objects/Geometry.php:
```diff
@@ -106,26 +107,41 @@ public static function fromWkt(string $wkt, int $srid = 0): static
+     *
+     * @throws OutOfBoundsException
+     */
+    public static function fromArray(array $geometry): static
```
Please add this method to the `API.md`

MatanYadaev commented on tests/Objects/GeometryCollectionTest.php:
```diff
@@ -52,22 +52,41 @@
+});
+
+it('creates geometry collection from array (serialized model)', function (): void {
+    $geometryCollection = new GeometryCollection([
+      new Polygon([
+        new LineString([
+          new Point(0, 180),
+          new Point(1, 179),
+          new Point(2, 178),
+          new Point(3, 177),
+          new Point(0, 180),
+        ]),
+      ]),
+      new Point(0, 180),
+    ]);
+
+    $geometryCollectionFromArray = GeometryCollection::fromArray($geometryCollection->toArray());
```
I tend to keep the tests following the [AAA pattern](https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F-12-structure-tests-by-the-aaa-pattern) - so anything that is related to arranging the data should be above the "act" part.
```suggestion
    $geometryCollection = new GeometryCollection([
      new Polygon([
        new LineString([
          new Point(0, 180),
          new Point(1, 179),
          new Point(2, 178),
          new Point(3, 177),
          new Point(0, 180),
        ]),
      ]),
      new Point(0, 180),
    ]);
    $geometryCollectionArray = $geometryCollection->toArray();

    $geometryCollectionFromArray = GeometryCollection::fromArray($geometryCollectionArray);
```

MatanYadaev commented on tests/Objects/GeometryCollectionTest.php:
```diff
@@ -52,22 +52,41 @@
+    expect($geometryCollectionFromJson)->toEqual($geometryCollection);
+});
+
+it('creates geometry collection from array (serialized model)', function (): void {
```
Please move the test to the `GeometryTest` because it's not related to a specific geometry type.

MatanYadaev reviewed (changes requested):
Please use 2 spaces instead of 4.

trin4ik commented:
- codestyle fixed
- throw `OutOfBoundsException` removed
- API documented
- test moved and fixed.

also added some test. How u run a tests? i use https://github.com/nektos/act and when i run pest/pest-coverage, i see a lot of coverage errors

![image](https://user-images.githubusercontent.com/839633/207049402-5e491bfe-03be-4869-b39e-9e2ab41a01bf.png)

MatanYadaev commented:
@trin4ik Thanks!
